src/lib/transport.js

```diff
@@ -37,17 +40,24 @@ function createPacketAndCondition ({
   const address = destinationAccount + '.~' + protocol + '.' + receiverId +
     (id ? ('.' + id) : '')
 
-  const blobData = omitUndefined({
-    expires_at: expiresAt,
-    data: data
-  })
+  const details = createDetails({
+    unsafeHeaders: Object.assign({
+      'Key-Id': receiverId, // TODO: is this right?
```

Comment:
`receiverId` should stay in the ILP address and the UUID that we were previously appending to the address should be moved to this header.

It makes sense for that header to be something like `Key: hmac-sha-256 a41d1a2f-632d-47b4-b730-3070270190ea` (or _maybe_ `Key-Gen`, since it's about how you generate the payment key). As @justmoon points out there's precedent for including two fields in the header value.

I think the thing after `hmac-sha-256` should be a nonce that the sender and recipient will use to generate the (unique) payment key (`HMAC(shared secret, nonce)`), which will be used for encryption and generating the fulfillment (`HMAC(payment key, ILP packet)`). We were already using it to ensure that the conditions were unique, so (I think) we might as well use a unique key for encryption as well.